uint16 red
uint16 green
uint16 blue


